src/features/pageObjects/pageObjList/pageObjListHeader/PageObjListHeader.tsx

```diff
@@ -21,17 +21,28 @@ const { confirm } = Modal;
 interface Props {
   template: Blob;
   toggleExpand: () => void;
+  setActivePanel: (pageObjectId: string[]) => void;
   isExpanded: boolean;
 }
 
-export const PageObjListHeader: React.FC<Props> = ({ template, toggleExpand, isExpanded }) => {
+export const PageObjListHeader: React.FC<Props> = ({ template, toggleExpand, isExpanded, setActivePanel }) => {
   const state = useSelector((state) => state) as RootState;
   const pageObjects = useSelector(selectPageObjects);
   const locatorsToGenerate = useSelector(selectLocatorsToGenerate);
   const enableDownload = useMemo(() => !!size(locatorsToGenerate), [locatorsToGenerate]);
+  const newPOstub = pageObjects.find((pageObject) => !pageObject.locators?.length);
+  const isSomePOcreated = pageObjects.some((pageObject) => pageObject.locators?.length);
 
   const dispatch = useDispatch();
-  const handleAddPageObject = () => dispatch(addPageObj());
+
+  const handleAddPageObject = () => {
+    if (newPOstub || (isSomePOcreated && newPOstub)) {
```

Comment:
I believe result will be same with `if (newPOstub) {`

Reply:
yes, of course.
Thank you!
fixed.